pytorch_lightning/trainer/trainer.py

```diff
@@ -520,6 +520,7 @@ def __init__(
 
         # override dist backend when using tpus
         if self.on_tpu:
+            self.distributed_backend = 'tpu'
```

Comment:
How about setting `self.distributed_backend = 'tpu'` when tpu_id is None since it doesn't trigger the spawn process when there is a tpu_id I think? Will it break anything? Also do this in `init_tpu()`??

Reply:
> How about setting self.distributed_backend = 'tpu' when tpu_id is None since it doesn't trigger the spawn process when there is a tpu_id I think? Will it break anything? 

I feel this will break as `transfer_distrib_spawn_state_on_fit_end` is being called even while training on individual TPU core here: https://github.com/PyTorchLightning/pytorch-lightning/blob/93ae175d51270b4a78a0cc1e74bdd4dd45c8791b/pytorch_lightning/accelerators/tpu_backend.py#L119

> Also do this in init_tpu()??

Discussed with @Borda about this. It's better to set the distributed_backend in trainer to maintain consistency.

Reply:
yeah, just checked. mp_queue will be None when there is a tpu_id.